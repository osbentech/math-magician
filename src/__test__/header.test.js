const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const { BrowserRouter: Router } = require('react-router-dom');
const Header = require('../header').default;

describe('Test the header component', () => {
  test('Renders header component correctly', () => {
    const header = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(header).toMatchSnapshot();
  });

});