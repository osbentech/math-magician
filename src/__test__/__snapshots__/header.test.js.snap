// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the header component Renders header component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          A Benjamin think...
        </h1>
        <ul>
          <a
            href="/"
          >
            Home
          </a>
          <span>
             | 
          </span>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <span>
             | 
          </span>
          <a
            href="/Quote"
          >
            Quotes
          </a>
        </ul>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        A Benjamin think...
      </h1>
      <ul>
        <a
          href="/"
        >
          Home
        </a>
        <span>
           | 
        </span>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        <span>
           | 
        </span>
        <a
          href="/Quote"
        >
          Quotes
        </a>
      </ul>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
